portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -183,19 +203,65 @@ private String getUploadedFileUrl(HttpServletRequest request, HttpServletRespons
     if (blobInfo.getContentType().contains("image") == false) {
       response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Only images allowed!");
       blobstoreService.delete(blobKey);
+      return null;
+    }
+
+    return blobKey;
+  }
+
+  /**
+   * Blobstore stores files as binary data. This function retrieves the binary data stored at the
+   * BlobKey parameter.
+   */
+  private byte[] getBlobBytes(BlobKey blobKey) throws IOException {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    ByteArrayOutputStream outputBytes = new ByteArrayOutputStream();
+
+    int fetchSize = BlobstoreService.MAX_BLOB_FETCH_SIZE;
+    long currentByteIndex = 0;
+    boolean continueReading = true;
+    while (continueReading) {
+      // end index is inclusive, so we have to subtract 1 to get fetchSize bytes
+      byte[] b =
+          blobstoreService.fetchData(blobKey, currentByteIndex, currentByteIndex + fetchSize - 1);
+      outputBytes.write(b);
+
+      // if we read fewer bytes than we requested, then we reached the end
```

Comment:
Please use capital case and full stop for comments which are sentences.

Reply:
Done